### DISCUSSION TITLE
dogecoin maybe need node auto version check and ban mechanism.

### DISCUSSION THREAD

**Discussion by dxteam2024** [dogecoin/dogecoin#2624] [General]
just like 1.14.4 , if anyone want to stop or slow down the whole net update in malicious , they just only need to deploy 1/2 old version nodes with less costs may about 1000 dollars per month? so if we have the ban mechanism: "auto kick out the nodes that have not been upgraded for a long time" something like that may be we can fix this "bug"!!

**Comment by dxteam2024** [comment]
as the node operator they have duty to update or do something.

**Comment by patricklodder** [comment]
We have this on the protocol. However, I wouldn't do it on the user agent string, as there is no value to it and you can change it. 

Instead of making this a "ban" thing, what we could do (experimentally) is make sure that we always connect to at least 2 nodes with a fee filter that is at least as permissive as ours. i.e. out of the 8 outgoing slots we have, whenever there aren't 2 peers with a fee filter equal to or lower than the one we set in that set, we keep on disconnecting 1 slot until we have connections established that will accept and relay our local policy. That way we have 6 peers that will serve us blocks and tx according to their policies, but we reserve 2 to get the policies we really need. 

Any thoughts?

**Reply by dxteam2024** [reply]
what if we want to add bigger block size or fasetr block out to the net？the fee filter strategy may not work

**Reply by patricklodder** [reply]
Those are protocol updates.

**Reply by MishaBoar** [reply]
> 
> Instead of making this a "ban" thing, what we could do (experimentally) is make sure that we always connect to at least 2 nodes with a fee filter that is at least as permissive as ours. i.e. out of the 8 outgoing slots we have, whenever there aren't 2 peers with a fee filter equal to or lower than the one we set in that set, we keep on disconnecting 1 slot until we have connections established that will accept and relay our local policy. That way we have 6 peers that will serve us blocks and tx according to their policies, but we reserve 2 to get the policies we really need.
> 
> Any thoughts?

As a node operator, I think this is a good idea. The fact that as a node operator I am choosing to run a specific version implies that I want to favor a certain policy and its survival - the one I picked for my own node. At the same time, leaving the 6 peers open makes sure we avoid creating enclaves/groups of nodes that become isolated and we make sure there is still competition between the different policies. As a node operator, I like it a lot - but of course I leave any considerations about consequences to the network to the experts.

**Reply by patricklodder** [reply]
> The fact that as a node operator I am choosing to run a specific version implies that I want to favor a certain policy and its survival - the one I picked for my own node.

I love this formulation, thank you for making it that crystal! If there's still time left after I finish my 1.14.5 cleanup run, I'll propose something. To be realistic, I don't have high hopes of this making it into 1.14.5 in time because it'll need far going scrutinization and testing, but this should definitely be possible for the next minor release.

**Comment by dxteam2024** [comment]
anyway we need some reasonable mechanism to keep the whole net updating in a predictable time ,otherwise doge net may face some "attack" with a low cost, and may slow down or destory the whole net update !

**Comment by victorsk2019** [comment]
Interesting discussion.  So is there a way for full node operators to somehow modify full node so as to "enforce" better lower transaction fees?  On my website www.stage4linux.com I make various RPC calls to get full node information, out of my personal curiosity:
![image](https://user-images.githubusercontent.com/47955466/144721962-03e84fd1-e2ba-4328-b9d8-8645155db3ce.png)

As the screenshot indicates, most transaction fees are still a bit (too) high.  My understanding is that in order for lower transaction fees to take effect, wallet users must upgrade their wallets to version 1.14.5.  Is there something else I can do to the full node on my end to ensure lower transaction fees go through?

**Reply by patricklodder** [reply]
> wallet users must upgrade their wallets to version 1.14.5 

That, or the equivalent of the lower fee version for the wallet software they use. There are many 3rd party wallets that have not implemented the lower fees yet.

> Is there something else I can do to the full node on my end to ensure lower transaction fees go through?

Your 1.14.5 node already allows these as a default policy out of the box, so if you don't change any of the relay policies, you're doing all you can on that front.
